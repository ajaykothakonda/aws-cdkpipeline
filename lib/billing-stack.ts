// import { Stack,  StackProps } from "aws-cdk-lib";
import { App } from "aws-cdk-lib";
import { Stack, StackProps } from 'aws-cdk-lib';
import { Budget } from "./constructs/budget";
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

interface BillingStackProps extends StackProps {

    budgetAmount: number
    emailAddress: string
}  

const app = new App()

export class BillingSTack extends Stack {
    constructor(scope: Construct, id: string, props: BillingStackProps) {

        super(scope, id, props);

        new Budget(this, 'Budget', {
            budgetAmount: props.budgetAmount,
           emailAddress: props.emailAddress
        })


    }
}
