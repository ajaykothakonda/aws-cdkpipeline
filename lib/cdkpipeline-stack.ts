import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Artifact, Pipeline } from 'aws-cdk-lib/aws-codepipeline'
import {aws_codepipeline} from "aws-cdk-lib";
import { GitHubSourceAction, CodeBuildAction } from "aws-cdk-lib/aws-codepipeline-actions";
import { SecretValue } from "aws-cdk-lib";
import {LinuxBuildImage, PipelineProject, BuildSpec} from "aws-cdk-lib/aws-codebuild";


// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class CdkpipelineStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const pipeline = new Pipeline(this, 'Pipeline', {
      pipelineName: 'Pipeline',
      crossAccountKeys: false
    })

    const sourceOutput = new Artifact('sourceOutput');

    pipeline.addStage({
      stageName: "source",
      actions: [
          new GitHubSourceAction({
            owner: 'kengne66',
            repo: 'aws-cdkpipeline',
            branch: 'master',
            actionName: 'Pipeline_Source',
            oauthToken: SecretValue.secretsManager('github-token'),
            output: sourceOutput
          })
      ]
    })

    const cdkBuildOutput = new Artifact('cdkBuildOutput')

    pipeline.addStage( {
      stageName: 'Build',
      actions: [
          new CodeBuildAction({
            actionName: "CDK_Build",
            input: sourceOutput,
            outputs: [cdkBuildOutput],
            project: new PipelineProject(this, 'cdkBuildProject', {
              environment: {
                buildImage: LinuxBuildImage.AMAZON_LINUX_2,
              },
              buildSpec: BuildSpec.fromSourceFilename("build-specs/cdk-build-spec"),


            })
          })

      ]
    })

    // The code that defines your stack goes here

    // example resource
    // const queue = new sqs.Queue(this, 'CdkpipelineQueue', {
    //   visibilityTimeout: cdk.Duration.seconds(300)
    // });
  }
}
