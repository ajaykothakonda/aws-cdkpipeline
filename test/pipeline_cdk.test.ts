import * as cdk from 'aws-cdk-lib';
// import { Template } from 'aws-cdk-lib/assertions';
import * as Cdkpipeline from '../lib/pipeline_cdk-stack';
import { App, Environment } from "aws-cdk-lib";
//import { expect as expectCDK, matchTemplate, MatchStyle, SynthUtils, } from "aws-cdk-lib/assets";
import { Capture, Match, Template } from "aws-cdk-lib/assertions";

import * as Pipeline from "../lib/pipeline_cdk-stack";
import { ServiceStack } from "../lib/service-stack";
import { PipelineCdkStack } from "../lib/pipeline_cdk-stack";
import { BillingSTack } from '../lib/billing-stack';

// example test. To run these tests, uncomment this file along with the
// example resource in lib/cdkpipeline-stack.ts
test('Pipeline Stack', () => {
   const app = new cdk.App();
//     // WHEN
   const stack1 = new Cdkpipeline.PipelineCdkStack(app, 'MyTestStack');
//     // THEN


   const template = Template.fromStack(stack1)
   expect(template.toJSON()).toMatchSnapshot()
});

test("Adding service stage", () => {
   // GIVEN
   const app = new App();
   const serviceStack = new ServiceStack(app, "ServiceStack", {
      stageName: "Test"
   });
   const pipelineStack = new PipelineCdkStack(app, "PipelineStack");
 
   // WHEN
   pipelineStack.addServiceStage(serviceStack, "Test");
 
   // THEN
   Template.fromStack(pipelineStack).hasResourceProperties(
     "AWS::CodePipeline::Pipeline",
     {
       Stages: Match.arrayWith([
         Match.objectLike({
           Name: "Test",
         }),
       ]),
     }
   );
 });

 test("Adding Billing stack test", () => {
   // GIVEN
   const app = new App();
   const serviceStack = new ServiceStack(app, "ServiceStack", {
      stageName: "Test"
   });
   const pipelineStack = new PipelineCdkStack(app, "PipelineStack");
   const billingStack = new BillingSTack(app, "billingStack", {
      budgetAmount: 5,
      emailAddress: "pierre_kengne@yahoo.com"
   });

   //const testStage =  pipelineStack.addServiceStage(serviceStack, "test");
 
   // WHEN
   //pipelineStack.addBillingStackToStage(serviceStack, testStage)
 
   // THEN
   Template.fromStack(pipelineStack).hasResourceProperties(
     "AWS::CodePipeline::Pipeline",
     {
       Stages: Match.arrayWith([
         Match.objectLike({
           Name: "Billing_Update",
         }),
       ]),
     }
   );
 });
 