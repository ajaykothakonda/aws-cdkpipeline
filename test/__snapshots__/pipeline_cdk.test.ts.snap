// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline Stack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "PipelineCDKArtifactsBucket388BB5A5": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "PipelineCDKArtifactsBucketPolicy1D6542A1": {
      "Properties": {
        "Bucket": {
          "Ref": "PipelineCDKArtifactsBucket388BB5A5",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineCDKArtifactsBucket388BB5A5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineCDKArtifactsBucket388BB5A5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "PipelineCDKBBF86984": {
      "DependsOn": [
        "PipelineCDKRoleDefaultPolicyFDF69877",
        "PipelineCDKRoleB1D32677",
      ],
      "Properties": {
        "ArtifactStore": {
          "Location": {
            "Ref": "PipelineCDKArtifactsBucket388BB5A5",
          },
          "Type": "S3",
        },
        "Name": "PipelineCDK",
        "RestartExecutionOnUpdate": true,
        "RoleArn": {
          "Fn::GetAtt": [
            "PipelineCDKRoleB1D32677",
            "Arn",
          ],
        },
        "Stages": [
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
                  "Owner": "kengne66",
                  "PollForSourceChanges": false,
                  "Repo": "aws-cdkpipeline",
                },
                "Name": "Pipeline_Source",
                "OutputArtifacts": [
                  {
                    "Name": "CDKsourceOutput",
                  },
                ],
                "RunOrder": 1,
              },
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
                  "Owner": "kengne66",
                  "PollForSourceChanges": false,
                  "Repo": "express-lambda",
                },
                "Name": "service_Source",
                "OutputArtifacts": [
                  {
                    "Name": "ServiceSourceOutput",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "ProjectName": {
                    "Ref": "cdkBuildProject9606D66E",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "CDKsourceOutput",
                  },
                ],
                "Name": "CDK_Build",
                "OutputArtifacts": [
                  {
                    "Name": "cdkBuildOutput",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "PipelineCDKBuildCDKBuildCodePipelineActionRole4D13E21D",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "ProjectName": {
                    "Ref": "ServiceBuildProject6D228AFA",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "ServiceSourceOutput",
                  },
                ],
                "Name": "Service_Build",
                "OutputArtifacts": [
                  {
                    "Name": "serviceBuildOutput",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "PipelineCDKBuildServiceBuildCodePipelineActionRole2C754962",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1",
                },
                "Configuration": {
                  "ActionMode": "CREATE_UPDATE",
                  "Capabilities": "CAPABILITY_NAMED_IAM",
                  "RoleArn": {
                    "Fn::GetAtt": [
                      "PipelineCDKpipelineUpdatePipelineUpdateRoleF9CBBC3B",
                      "Arn",
                    ],
                  },
                  "StackName": "PipelineCDKStack",
                  "TemplatePath": "cdkBuildOutput::PipelineCDKStack.template.json",
                },
                "InputArtifacts": [
                  {
                    "Name": "cdkBuildOutput",
                  },
                ],
                "Name": "Pipeline_Update",
                "RoleArn": {
                  "Fn::GetAtt": [
                    "PipelineCDKpipelineUpdatePipelineUpdateCodePipelineActionRoleF76A1162",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "pipeline_Update",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "PipelineCDKBuildCDKBuildCodePipelineActionRole4D13E21D": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineCDKBuildCDKBuildCodePipelineActionRoleDefaultPolicyF7287F3B": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "cdkBuildProject9606D66E",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineCDKBuildCDKBuildCodePipelineActionRoleDefaultPolicyF7287F3B",
        "Roles": [
          {
            "Ref": "PipelineCDKBuildCDKBuildCodePipelineActionRole4D13E21D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineCDKBuildServiceBuildCodePipelineActionRole2C754962": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineCDKBuildServiceBuildCodePipelineActionRoleDefaultPolicyFB7E0C31": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "ServiceBuildProject6D228AFA",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineCDKBuildServiceBuildCodePipelineActionRoleDefaultPolicyFB7E0C31",
        "Roles": [
          {
            "Ref": "PipelineCDKBuildServiceBuildCodePipelineActionRole2C754962",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineCDKRoleB1D32677": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineCDKRoleDefaultPolicyFDF69877": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineCDKArtifactsBucket388BB5A5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineCDKArtifactsBucket388BB5A5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineCDKBuildCDKBuildCodePipelineActionRole4D13E21D",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineCDKBuildServiceBuildCodePipelineActionRole2C754962",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineCDKpipelineUpdatePipelineUpdateCodePipelineActionRoleF76A1162",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineCDKRoleDefaultPolicyFDF69877",
        "Roles": [
          {
            "Ref": "PipelineCDKRoleB1D32677",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineCDKSourcePipelineSourceWebhookResource207275C3": {
      "Properties": {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": {
          "SecretToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
        },
        "Filters": [
          {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "Pipeline_Source",
        "TargetPipeline": {
          "Ref": "PipelineCDKBBF86984",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
    "PipelineCDKSourceserviceSourceWebhookResourceBE4CE570": {
      "Properties": {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": {
          "SecretToken": "{{resolve:secretsmanager:github-token:SecretString:::}}",
        },
        "Filters": [
          {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "service_Source",
        "TargetPipeline": {
          "Ref": "PipelineCDKBBF86984",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
    "PipelineCDKpipelineUpdatePipelineUpdateCodePipelineActionRoleDefaultPolicy5CC061E6": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "iam:PassRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineCDKpipelineUpdatePipelineUpdateRoleF9CBBC3B",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineCDKArtifactsBucket388BB5A5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineCDKArtifactsBucket388BB5A5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "cloudformation:CreateStack",
                "cloudformation:DescribeStack*",
                "cloudformation:GetStackPolicy",
                "cloudformation:GetTemplate*",
                "cloudformation:SetStackPolicy",
                "cloudformation:UpdateStack",
                "cloudformation:ValidateTemplate",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":cloudformation:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stack/PipelineCDKStack/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineCDKpipelineUpdatePipelineUpdateCodePipelineActionRoleDefaultPolicy5CC061E6",
        "Roles": [
          {
            "Ref": "PipelineCDKpipelineUpdatePipelineUpdateCodePipelineActionRoleF76A1162",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineCDKpipelineUpdatePipelineUpdateCodePipelineActionRoleF76A1162": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineCDKpipelineUpdatePipelineUpdateRoleDefaultPolicy1A5BB9DA": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineCDKArtifactsBucket388BB5A5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineCDKArtifactsBucket388BB5A5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "*",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PipelineCDKpipelineUpdatePipelineUpdateRoleDefaultPolicy1A5BB9DA",
        "Roles": [
          {
            "Ref": "PipelineCDKpipelineUpdatePipelineUpdateRoleF9CBBC3B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PipelineCDKpipelineUpdatePipelineUpdateRoleF9CBBC3B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "cloudformation.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineNotificationsTopic827A419F": {
      "Properties": {
        "TopicName": "PipelineNotificationsTopic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "PipelineNotificationsTopicpierrekengneyahoocom17355058": {
      "Properties": {
        "Endpoint": "pierre_kengne@yahoo.com",
        "Protocol": "email",
        "TopicArn": {
          "Ref": "PipelineNotificationsTopic827A419F",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "ServiceBuildProject6D228AFA": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "ServiceBuildProjectRoleEB246009",
            "Arn",
          ],
        },
        "Source": {
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "ServiceBuildProjectRoleDefaultPolicy29400A8A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "ServiceBuildProject6D228AFA",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "ServiceBuildProject6D228AFA",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "ServiceBuildProject6D228AFA",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineCDKArtifactsBucket388BB5A5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineCDKArtifactsBucket388BB5A5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ServiceBuildProjectRoleDefaultPolicy29400A8A",
        "Roles": [
          {
            "Ref": "ServiceBuildProjectRoleEB246009",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ServiceBuildProjectRoleEB246009": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "cdkBuildProject9606D66E": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "cdkBuildProjectRoleD43CAD48",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":s3:::",
                {
                  "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                },
                "/3b3c9284a2ff3bfd73210f6b0c3dbb48201ca20c21081cd5275d3fe339544793.yml",
              ],
            ],
          },
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "cdkBuildProjectRoleD43CAD48": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "cdkBuildProjectRoleDefaultPolicy030C433D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "cdkBuildProject9606D66E",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "cdkBuildProject9606D66E",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "cdkBuildProject9606D66E",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineCDKArtifactsBucket388BB5A5",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineCDKArtifactsBucket388BB5A5",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "cdkBuildProjectRoleDefaultPolicy030C433D",
        "Roles": [
          {
            "Ref": "cdkBuildProjectRoleD43CAD48",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
